source/common/router/router.cc

```diff
@@ -124,7 +124,17 @@ uint64_t FilterUtility::percentageOfTimeout(const std::chrono::milliseconds resp
   return static_cast<uint64_t>(response_time.count() * TimeoutPrecisionFactor / timeout.count());
 }
 
-void FilterUtility::setUpstreamScheme(Http::RequestHeaderMap& headers, bool downstream_secure) {
+void FilterUtility::setUpstreamScheme(Http::RequestHeaderMap& headers, bool downstream_secure,
```

Comment:
so there's a difference in what we pass in here
I think we need to decide if we want to set scheme based on "is upstream ssl" or "is upstream secure"
e.g. ALTS is secure but not TLS

Reply:
Looking at the [golang grpc](https://github.com/grpc/grpc-go/blob/0020ccf9d9fc448fcfa099173df4ad330e96326c/internal/transport/http2_client.go#L304-L306) and [alts](https://github.com/grpc/grpc-go/blob/master/credentials/alts/alts.go#L160-L163) source code, it seems like this should only be set when the upstream is using TLS and not ATLS. The [RFC](https://www.rfc-editor.org/rfc/rfc9110.html#name-https-uri-scheme) also seems to couple it with TLS.

Reply:
Decided to keep it simple and just check if it has an SSL context (TLS) to set the scheme to https, otherwise use http.